File: Data Structures & Algorithms/three-integer-sum/submission-2.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        answer = []
        nums.sort()

        for index, val in enumerate(nums):
            if val > 0:
                break
            
            if index > 0 and val == nums[index - 1]:
                continue
            
            left, right = index + 1, len(nums) - 1

            while left < right:
                threeSum = val + nums[left] + nums[right]
                if threeSum > 0:
                    right -= 1
                elif threeSum < 0:
                    left += 1
                else:
                    answer.append([val, nums[left], nums[right]])
                    left += 1
                    right -= 1
                    while nums[left] == nums[left - 1] and left < right:
                        left += 1
        return answer

File: Data Structures & Algorithms/contains-duplicate-ii/submission-2.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        for(int left = 0; left < nums.length; left++) {
            for(int right = left + 1; right < Math.min(nums.length, left + k + 1); right++) {
                if (nums[left] == nums[right]) {
                    return true;
                }
            }
        }
        return false;
    }
}